src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/AuthorisedCaseDetailsSearchOperation.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.ccd.domain.service.search.elasticsearch;
+
+import static java.util.Collections.emptyList;
+import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseType;
+import uk.gov.hmcts.ccd.domain.model.search.CaseDetailsSearchResult;
+import uk.gov.hmcts.ccd.domain.service.common.AuthorisedCaseDataFilter;
+import uk.gov.hmcts.ccd.domain.service.common.AuthorisedCaseDefinitionDataService;
+
+@Service
+@Qualifier(AuthorisedCaseDetailsSearchOperation.QUALIFIER)
+public class AuthorisedCaseDetailsSearchOperation implements CaseDetailsSearchOperation {
+
+    public static final String QUALIFIER = "AuthorisedCaseDetailsSearchOperation";
+
+    private final CaseDetailsSearchOperation caseDetailsSearchOperation;
+    private final AuthorisedCaseDefinitionDataService authorisedCaseDefinitionDataService;
+    private final AuthorisedCaseDataFilter authorisedCaseDataFilter;
+
+    public AuthorisedCaseDetailsSearchOperation(
+        @Qualifier(ElasticsearchCaseDetailsSearchOperation.QUALIFIER) CaseDetailsSearchOperation caseDetailsSearchOperation,
+        AuthorisedCaseDefinitionDataService authorisedCaseDefinitionDataService,
+        AuthorisedCaseDataFilter authorisedCaseDataFilter) {
+
+        this.caseDetailsSearchOperation = caseDetailsSearchOperation;
+        this.authorisedCaseDefinitionDataService = authorisedCaseDefinitionDataService;
+        this.authorisedCaseDataFilter = authorisedCaseDataFilter;
+    }
+
+    @Override
+    public CaseDetailsSearchResult execute(String caseTypeId, String query) {
```

Comment:
my mistake. You mind renaming this parameter from query to jsonSearchRequest or searchRequest?  we pass through the whole search request not only the query part
Can you rename also in the interface please

Reply:
renamed to jsonQuery

Reply:
the point is that jsonSearchRequest and jsonQuery are 2 different things. From the controller we receive a jsonSearchRequest, which contains a query, size, from.... properties. 
This parameter is the whole request, not only the query

Reply:
Introduced new object CaseSearchRequest that converts Json string to JsonNode and validates existence of query field.